internal/client/v1/client/agent/core/client.go

```diff
@@ -126,3 +175,59 @@ func (c *agentClient) IndexInfo(
 	}
 	return res, nil
 }
+
+func (c *singleAgentClient) CreateIndex(
```

Comment:
should we separate the implementation of singleAgentClient to another file?

Reply:
why?

Reply:
for maintainability / readability 🤔

Reply:
for maintainability, I think it would better to write them in single file.

Reply:
Maintainability and readability are not always in the same place.